### PR TITLE
feat(transformImport): support disable default transform import

### PR DESCRIPTION
**Pull request by JSerFeng**
## Summary

Support using `transformImport: false` to disable default options for `antd`

### <samp>🤖 Generated by Copilot at db55714</samp>

This pull request enhances the transformImport feature of the modern.js builder, by allowing users to disable it or customize it for different import scenarios. It also adds antd UI library support for the rspack and webpack bundlers. It updates the types, plugins, and tests files accordingly. It also adds a changeset file and peer dependencies for the release process.

### <samp>🤖 Generated by Copilot at db55714</samp>

*  Add a changeset file to describe the version bumps and commit messages ([link](https://github.com/web-infra-dev/modern.js/pull/3511/files?diff=unified&w=0#diff-8a97fb1711044859ce7f64569b9e19fbc9f0bfc166cf102c7ce5b70560ef14c7R1-R11))
*  Add react and react-dom as peer dependencies for the builder-rspack-provider package ([link](https://github.com/web-infra-dev/modern.js/pull/3511/files?diff=unified&w=0#diff-0dfad59d3b9faa8b69b516527ff7dcc43db019040e3499dfa12a7b716d2e7ab7L58-R60))
*  Allow the transformImport option to accept a false value to disable the default import transformation behavior for the rspack, webpack, and babel bundlers ([link](https://github.com/web-infra-dev/modern.js/pull/3511/files?diff=unified&w=0#diff-40472263d401ace07efe02124aba4a15a2890aec689e94f1f13868aec4d94a1aL138-R140), [link](https://github.com/web-infra-dev/modern.js/pull/3511/files?diff=unified&w=0#diff-1d881eb21400957a6f782153fd8eba41c23944a2c91461b579296019ece0578aL206-R208), [link](https://github.com/web-infra-dev/modern.js/pull/3511/files?diff=unified&w=0#diff-4b0b13aed01ac9bd91f13a1dedc01fc1da2cf2d006cb8f710b43dbcac770e95aL32-R32), [link](https://github.com/web-infra-dev/modern.js/pull/3511/files?diff=unified&w=0#diff-4b0b13aed01ac9bd91f13a1dedc01fc1da2cf2d006cb8f710b43dbcac770e95aL44-R44), [link](https://github.com/web-infra-dev/modern.js/pull/3511/files?diff=unified&w=0#diff-266a128dbe0dbe9468a8e5c2a951307124a94d579b3dee10683973ffa475ad9cR15))
*  Update the source configuration types to allow the transformImport option to accept a false value ([link](https://github.com/web-infra-dev/modern.js/pull/3511/files?diff=unified&w=0#diff-bc0f240d77e7746d24f91bc3027bffb0509e8c150f8fab12788df0142300b3abL10-R15), [link](https://github.com/web-infra-dev/modern.js/pull/3511/files?diff=unified&w=0#diff-5549f76a51d840d8670e1a2f11eec13f3ab37e9f0cbddaa8ca9ed3e68f2b0a05L59-R59), [link](https://github.com/web-infra-dev/modern.js/pull/3511/files?diff=unified&w=0#diff-5549f76a51d840d8670e1a2f11eec13f3ab37e9f0cbddaa8ca9ed3e68f2b0a05R67-R68))
*  Add two new properties to the transformImport option to customize the import transformation for different naming conventions and import types ([link](https://github.com/web-infra-dev/modern.js/pull/3511/files?diff=unified&w=0#diff-5549f76a51d840d8670e1a2f11eec13f3ab37e9f0cbddaa8ca9ed3e68f2b0a05R67-R68))
*  Add test cases for the swc and webpack plugins to check the pluginImport option with the builderAntdPlugin and the false value ([link](https://github.com/web-infra-dev/modern.js/pull/3511/files?diff=unified&w=0#diff-6300f5d547d33eaf5fd9e5f37d0f787a8ce36658d273c1d861ea601b485ee756R7), [link](https://github.com/web-infra-dev/modern.js/pull/3511/files?diff=unified&w=0#diff-6300f5d547d33eaf5fd9e5f37d0f787a8ce36658d273c1d861ea601b485ee756R94-R134), [link](https://github.com/web-infra-dev/modern.js/pull/3511/files?diff=unified&w=0#diff-94bad7f67a9adfac79c653b8f93c38977b1ae06fa8cc2f3d601c84c85afa8cc3R5), [link](https://github.com/web-infra-dev/modern.js/pull/3511/files?diff=unified&w=0#diff-94bad7f67a9adfac79c653b8f93c38977b1ae06fa8cc2f3d601c84c85afa8cc3R229-R265))
*  Add a mock foo package to test the transformImport option with different customName and template configurations ([link](https://github.com/web-infra-dev/modern.js/pull/3511/files?diff=unified&w=0#diff-9514a0b3773f5e46121ed7f130583cc4bb9cc1894c3bc89e2777524de2b229cc), [link](https://github.com/web-infra-dev/modern.js/pull/3511/files?diff=unified&w=0#diff-afea46fa1d063d9f88e22cd126e32a6f067bbe2759f828256b1c143e01d3089aR1), [link](https://github.com/web-infra-dev/modern.js/pull/3511/files?diff=unified&w=0#diff-54471c23672e1cd283a7768dc2c291011feef789dc329411e1a4950a40a77e80R1), [link](https://github.com/web-infra-dev/modern.js/pull/3511/files?diff=unified&w=0#diff-65325452fa24f75f9e0d314d40a907b72c0154aab00d795a6369a5dc0c0cbfb4R1))
*  Update the plugin-import test file to use the mock foo package and add test cases for different bundlers and plugins with the same transformImport configuration ([link](https://github.com/web-infra-dev/modern.js/pull/3511/files?diff=unified&w=0#diff-8d377dd5181442e2e489e09dbe697ea95bd979e8cc5ec769990ecedcd729cf85L6-R13), [link](https://github.com/web-infra-dev/modern.js/pull/3511/files?diff=unified&w=0#diff-8d377dd5181442e2e489e09dbe697ea95bd979e8cc5ec769990ecedcd729cf85L18-R24), [link](https://github.com/web-infra-dev/modern.js/pull/3511/files?diff=unified&w=0#diff-8d377dd5181442e2e489e09dbe697ea95bd979e8cc5ec769990ecedcd729cf85L29-R87), [link](https://github.com/web-infra-dev/modern.js/pull/3511/files?diff=unified&w=0#diff-8d377dd5181442e2e489e09dbe697ea95bd979e8cc5ec769990ecedcd729cf85L82-R99), [link](https://github.com/web-infra-dev/modern.js/pull/3511/files?diff=unified&w=0#diff-8d377dd5181442e2e489e09dbe697ea95bd979e8cc5ec769990ecedcd729cf85L98-R115), [link](https://github.com/web-infra-dev/modern.js/pull/3511/files?diff=unified&w=0#diff-8d377dd5181442e2e489e09dbe697ea95bd979e8cc5ec769990ecedcd729cf85L110-R147), [link](https://github.com/web-infra-dev/modern.js/pull/3511/files?diff=unified&w=0#diff-8d377dd5181442e2e489e09dbe697ea95bd979e8cc5ec769990ecedcd729cf85R154-R220))
*  Add new entry files to test the import with different options ([link](https://github.com/web-infra-dev/modern.js/pull/3511/files?diff=unified&w=0#diff-f5fe620926790187dc22de772bd0c9b5c22bdd985e9bab394613bab98969ea5eR1-R3), [link](https://github.com/web-infra-dev/modern.js/pull/3511/files?diff=unified&w=0#diff-85a45295978bd1b85190f05b45d7c54d68db05315290f364716ee29eff6c9edbL1-R1), [link](https://github.com/web-infra-dev/modern.js/pull/3511/files?diff=unified&w=0#diff-fc4fbb0dd447258661a87928e246d50afef1bfd2d8d87f99449d0b16c0bae4bcR1-R3), [link](https://github.com/web-infra-dev/modern.js/pull/3511/files?diff=unified&w=0#diff-33f39e3d1311c9e64ec53e1a34dff11907c0519d0063a6e8ac05890deaae62e6R1-R3))

## Checklist

- [x] I have added changeset via `pnpm run change`.
- [x] I have updated the documentation.
- [x] I have added tests to cover my changes.

### CODE REVIEW

File: packages/builder/builder-rspack-provider/src/types/config/source.ts [hunk 1]
```diff
@@ -7,10 +7,10 @@ import type { RspackBuiltinsConfig } from '../rspack';
 
 export type SourceConfig = SharedSourceConfig & {
   define?: RspackBuiltinsConfig['define'];
-  transformImport?: Builtins['pluginImport'];
+  transformImport?: false | Builtins['pluginImport'];
```
Comment (chenjiahan) [inline packages/builder/builder-rspack-provider/src/types/config/source.ts:10]: It seems that `SharedSourceConfig` already provides the `transformImport` type

File: packages/document/builder-doc/docs/en/config/source/transformImport.md [hunk 2]
```diff
@@ -5,7 +5,7 @@ The difference from [babel-plugin-import](https://www.npmjs.com/package/babel-pl
 - **Type:**
 
 ```ts
-Array<{
+export type Config = false | Array<{
```
Comment (chenjiahan) [inline packages/document/builder-doc/docs/en/config/source/transformImport.md:8]:
```suggestion
type Config = false | Array<{
```

File: packages/document/builder-doc/docs/zh/config/source/transformImport.md [hunk 3]
```diff
@@ -5,7 +5,7 @@
 - **类型：**
 
 ```ts
-Array<{
+export type Config = false | Array<{
```
Comment (chenjiahan) [inline packages/document/builder-doc/docs/zh/config/source/transformImport.md:8]:
```suggestion
type Config = false | Array<{
```

File: packages/document/builder-doc/docs/zh/config/source/transformImport.md [hunk 4]
```diff
@@ -29,6 +29,8 @@ Array<{
 }
 ```
 
+注意如果你使用 `external` 将 antd 不打包进产物中，那么上述的默认行为会导致该 antd 的引入发生改变，从而使 external 不生效，你可以手动设置 `transformImport: false` 来关掉 transformImport 的默认行为。
```
Comment (chenjiahan) [inline packages/document/builder-doc/docs/zh/config/source/transformImport.md:32]: 这里不一定只有 external 的场景需要禁用 transformImport，所以描述顺序可以调整一下。

```suggestion
你可以手动设置 `transformImport: false` 来关掉 transformImport 的默认行为。

比如，当你使用了 `externals` 来避免打包 antd 时，由于 `transformImport` 默认会转换 antd 的引用路径，导致匹配的路径发生了变化，因此 externals 无法生效。此时你可以设置 `transformImport: false` 来避免该问题。
```

Comment (chenjiahan) [reply packages/document/builder-doc/docs/zh/config/source/transformImport.md:32]: 英文文档也类似

Comment (JSerFeng) [reply packages/document/builder-doc/docs/zh/config/source/transformImport.md]: Done

**Review by chenjiahan — APPROVED** [review]

**Pull request merged**